Userland/Libraries/LibAudio/QOATypes.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "QOATypes.h"
+#include <AK/Endian.h>
+#include <AK/Stream.h>
+
+namespace Audio::QOA {
+
+ErrorOr<FrameHeader> FrameHeader::read_from_stream(Stream& stream)
+{
+    FrameHeader header;
+    header.num_channels = TRY(stream.read_value<u8>());
+    u8 sample_rate[3];
+    // Enforce the order of the reads here, since the order of expression evaluations further down is implementation-defined.
+    sample_rate[0] = TRY(stream.read_value<u8>());
+    sample_rate[1] = TRY(stream.read_value<u8>());
+    sample_rate[2] = TRY(stream.read_value<u8>());
+    header.sample_rate = (sample_rate[0] << 16) | (sample_rate[1] << 8) | sample_rate[2];
+    header.sample_count = TRY(stream.read_value<BigEndian<u16>>());
+    header.frame_size = TRY(stream.read_value<BigEndian<u16>>());
+    return header;
+}
+
+LMSState::LMSState(u64 history_packed, u64 weights_packed)
+{
+    for (size_t i = 0; i < lms_history; ++i) {
```

Comment:
Could you either also use `QOA::lms_history` or change the other references to not have `QOA` for consistency?